src/test/java/uk/gov/hmcts/reform/civil/service/stitching/CivilDocumentStitchingServiceTest.java

```diff
@@ -0,0 +1,128 @@
+
+@ExtendWith(SpringExtension.class)
+@SpringBootTest(classes = {
+    CivilDocumentStitchingService.class,
+    JacksonAutoConfiguration.class,
+    CaseDetailsConverter.class
+})
+public class CivilDocumentStitchingServiceTest {
+    @Autowired
+    private CivilDocumentStitchingService civilDocumentStitchingService;
+    @MockBean
+    private BundleRequestExecutor bundleRequestExecutor;
+    @MockBean
+    private StitchingConfiguration stitchingConfiguration;
+    private static final CaseDocument CLAIM_FORM =
+        CaseDocument.builder()
+            .createdBy("John")
+            .documentName(String.format(N1.getDocumentTitle(), "000DC001"))
+            .documentSize(0L)
+            .documentType(SEALED_CLAIM)
+            .createdDatetime(LocalDateTime.now())
+            .documentLink(Document.builder()
+                              .documentUrl("fake-url")
+                              .documentFileName("file-name")
+                              .documentBinaryUrl("binary-url")
+                              .build())
+            .build();
+    //Optional should be removed after refactoring
+    private final Optional<String> stitchStatus = Optional.of(randomAlphabetic(8));
+    private final Optional<Document> stitchedDocument = Optional.of(mock(Document.class));
+
+    private List<DocumentMetaData> buildDocumentMetaDataList() {
+        return List.of(
+            new DocumentMetaData(
+                CLAIM_FORM.getDocumentLink(),
+                "Sealed Claim Form",
+                LocalDate.now().toString()
+            )
+        );
+    }
+
+    public CaseData buildCaseData(Bundle bundle){
+        List<IdValue<Bundle>> caseBundles = List.of(new IdValue<>("1", bundle));
+        return CaseDataBuilder.builder().caseBundles(caseBundles).build();
+    }
+    @Test
+    public void whenCaseDataIsNullWithPayloadAndNoCaseBundlesNoStitchingDocument() {
+        List<DocumentMetaData> documentMetaDataList = new ArrayList<>();
+        CaseData caseData = CaseDataBuilder.builder().legacyCaseReference("ClaimNumber").build();
+        CaseDocument caseDocument = civilDocumentStitchingService.bundle(documentMetaDataList, "Auth",
+                                                                  "Title", "FileName", caseData);
+        Assertions.assertNull(caseDocument);
+    }
+    @Test
+    public void whenCaseDataIsNullWithNoCaseBundles() {
+        List<DocumentMetaData> documentMetaDataList = buildDocumentMetaDataList();
+        CaseData caseData = CaseDataBuilder.builder().legacyCaseReference("ClaimNumber").build();
+        when(stitchingConfiguration.getStitchingUrl()).thenReturn("dummy_url");
+        CaseDocument caseDocument = civilDocumentStitchingService.bundle(documentMetaDataList, "Auth",
+                                                                         "Title", "FileName", caseData);
+        Assertions.assertNull(caseDocument);
+    }
```

Comment:
Put `// Given`, `// When`, `// Then` comments in all your tests to divide the three stages of each test. Better if along with them you clarify what's in each section. Example:
```
// Given: a case data with no case bundles
```

Chances are you'll also end up changing the `when(sticthing[...]` to a `given(stitching[...]`.